src/Adapters/MockAdapter.ts

```diff
@@ -1049,7 +1052,17 @@ export default class MockAdapter
         useStaticData?: boolean
     ) {
         let mockData: Array<ADXTimeSeries> = [];
-        let mockTimeSeriesData: Array<Array<TimeSeriesData>> = this.mockData;
+        let mockTimeSeriesData: Array<Array<TimeSeriesData>>;
+        if (
+            this.mockData &&
+            Array.isArray(this.mockData) &&
+            this.mockData.every(
+                (arr) => Array.isArray(arr) && arr.every(isTimeSeriesData)
+            )
+        ) {
+            mockTimeSeriesData = this.mockData;
```

Comment:
why not just use the mockData object in the mocked out adapter call? Do we need multiple sets of mocked data in the same test?

Reply:
mockData is a any type of value that you can set for MockAdapter. As you can see in the screenshot above, for ADT3DScenePage the mockData is set to a mock config object. So it is hard to structure the mockData type now, e.g. the mock time series is not a part of the mockData object that we send to the adapter. Please check out my screenshots above to see the issue.